### ISSUE TITLE
How to attch files using the Richtext Component

### ISSUE THREAD

**Issue by chavu** [rjaros/kvision#519]
I added a Richtext element to a form. It has an 'Attach files' action button. It alows use to pick a file to attach. How do I get hold of the picked file and send it to the save for saving in DB or file system? And also to show a link in the Richtext field content.

**Comment by rjaros** [comment]
This is not supported currently. It should be possible according to the docs: https://github.com/basecamp/trix?tab=readme-ov-file#storing-attached-files, but I would have to play a bit with this. I'll leave this issue open and get back when I have more info.

**Comment by chavu** [comment]
Thank you for the response.  I have noticed also that it doesn't support tables. Is there a way to use other editors that support tables?

**Comment by rjaros** [comment]
In general, yes. You can write wrappers for any JS library out there (e.g CKEditor, Tiny MCE and others).  But it's not an easy task because you need to know the chosen library quite well. As an alternative you can try using some React component, which would probably be easier to adapt but with additional bundle size cost. You can find an example of using editor-type React component in the guide: https://kvision.gitbook.io/kvision-guide/4.-integrating-with-javascript-libraries/using-react-components#advanced-components

**Comment by rjaros** [comment]
I've unblocked file uploading with a new `RichText`/`RichTextInput` parameter. In the next release you will be able to process attachments like this:
```kotlin
richText(allowFileUploads = true) {
    onEvent {
        event("trix-attachment-add") { event: dynamic ->
            if (event.attachment.file != null) {
                KVScope.launch { // we will use suspending function so lauch is needed 
                    val content = event.attachment.file.unsafeCast<File>().getContent() // extension function from io.kvision.utils
                    console.log(content) // content is a String containing data uri
                    // save content somewhere on a server, get url in return
                    val fileUrl = uploadService.uploadFileAndReturnUrl(content)
                    event.attachment.setAttributes(
                        obj {
                            this.url = fileUrl   // change file url in the edited html code
                        }
                    )
                }
            }
        }
    }
}
```

**Comment by chavu** [comment]
THisis good news. When is the next release which I can start using this code?

**Comment by rjaros** [comment]
It's being published right now.

**Comment by rjaros** [comment]
Fixed in 7.4.3
